### PR TITLE
Tools: Update cyber data extraction tools. major update on data_extract.py

### PR DESCRIPTION
**Pull request by gchen-apollo**
(1)Redefine logistics about channel message extraction.
(2)Enable multi-channel extraction.
(3)Fix bugs about storage criterion.
(4)Switch from opencv1 to opencv2 in Mat handling.:w

### CODE REVIEW

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py [hunk 1]
```diff
@@ -62,57 +63,47 @@ def process_dir(path, operation):
...
-                return False
-        elif image.encoding == "gray" or image.encoding == "y":
-            if image.step != image.width:
-                print("Image.step %d does not equal Image.width %d or gray image" %
-                    (image.step, image.width))
-                return False
-        else:
-            print("Unsupported image encoding type %s" %encoding)
+    cur_time_second = msg.timestamp
+    image = IMAGE_OBJ
```
Comment (xiaoxq) [inline modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: Is it intended to reuse the image instance every time?

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py [hunk 2]
```diff
@@ -277,23 +309,27 @@ def main():
...
+    parser.add_argument("-c", "--channel_name", dest='channel_list', action="append",
+                        default=[], help="list of channel_name that needs parsing.")
+    parser.add_argument("-s", "--start_timestamp", action="store", type=float,
+                        default=np.finfo(np.float32).min, help="Specify the begining time to extract data information.")
+    parser.add_argument("-e", "--end_timestamp", action="store", type=float,
+                        default=np.finfo(np.float32).max, help="Specify the ending timestamp to extract data information.")
+    parser.add_argument("-r", "--extraction_ratio", action="store", type=int,
                         default=10, help="The output compressed file.")
-
+
```
Comment (xiaoxq) [inline modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: Remove trailing spaces.

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py (hunk 1 above)
Comment (gchen-apollo) [reply modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: yes. create image instance every time might cause time cost. In the future, will put the functions in a class and image instance as a member.

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py (hunk 2 above)
Comment (gchen-apollo) [reply modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: solved

**Pull request merged**